integration_tests/cork_test.go

```diff
@@ -0,0 +1,257 @@
+package integration_tests
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"strings"
+	"time"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	allocationTypes "github.com/peggyjv/sommelier/v3/x/allocation/types"
+	corkTypes "github.com/peggyjv/sommelier/v3/x/cork/types"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const CellarABISubset = `
+[
+	{
+		"inputs": [
+		  {
+			"components": [
+			  {
+				"internalType": "uint184",
+				"name": "tokenId",
+				"type": "uint184"
+			  },
+			  {
+				"internalType": "int24",
+				"name": "tickUpper",
+				"type": "int24"
+			  },
+			  {
+				"internalType": "int24",
+				"name": "tickLower",
+				"type": "int24"
+			  },
+			  {
+				"internalType": "uint24",
+				"name": "weight",
+				"type": "uint24"
+			  }
+			],
+			"internalType": "struct ICellarPoolShare.CellarTickInfo[]",
+			"name": "_cellarTickInfo",
+			"type": "tuple[]"
+		  },
+		  {
+			"internalType": "uint256",
+			"name": "currentPriceX96",
+			"type": "uint256"
+		  }
+		],
+		"name": "rebalance",
+		"outputs": [],
+		"stateMutability": "nonpayable",
+		"type": "function"
+	  }
+]
+`
+
+type ABIEncodedTickRange struct {
+	TokenID *big.Int `abi:"tokenId"`
+	Upper   *big.Int `abi:"tickUpper"`
+	Lower   *big.Int `abi:"tickLower"`
+	Weight  *big.Int `abi:"weight"`
+}
+
+func ABIEncodedRebalance() []byte {
+	encodedCall, err := abi.JSON(strings.NewReader(CellarABISubset))
+	if err != nil {
+		panic(sdkerrors.Wrap(err, "bad ABI definition in code"))
+	}
+	args := []ABIEncodedTickRange{
+		{
+			TokenID: big.NewInt(int64(0)),
```

Comment:
Does it matter whether or not the source is a `uint` here? Are the input int sizes not relevant when instantiating `big.NewInt`?